### PR TITLE
fix(patch): allow @Equatable on types without stored properties

### PR DESCRIPTION
**Pull request by supersonicbyte**
## Description

Allow `@Equatable` to be applied on types without stored properties.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. 

## Minimal checklist:

- [x] I have performed a self-review of my own code 
- [x] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [x] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by calda** [comment]
Probably also want a test case that it generates a valid but empty `==` implementation

File: Tests/EquatableMacroTests.swift [hunk 1]
```diff
@@ -482,27 +478,65 @@ struct EquatableMacroTests {
...
+            struct NoProperties: View, Hashable {
+                let onTap: () -> Void
+
+                var body: some View {
+                    Text("")
+                }
+            }
+
+            extension NoProperties: Equatable {
+                nonisolated public static func == (lhs: NoProperties, rhs: NoProperties) -> Bool {
```
Comment (supersonicbyte) [inline Tests/EquatableMacroTests.swift:530]: @calda already there
